src/generic/grid.cpp

```diff
@@ -3859,11 +3862,75 @@ void wxGrid::PrepareDCFor(wxDC &dc, wxGridWindow *gridWindow)
     dc.SetDeviceOrigin(dcOrigin.x, dcOrigin.y);
 }
 
+void CheckDoDragScroll(wxGrid *grid, wxGridSubwindow *eventGridWindow,
+    wxGridSubwindow *gridWindow, wxPoint event_pos,
+    bool horizontal, bool vertical)
+{
+    // scroll when at the edges or outside the window
+    // eventGridWindow: the window that received the mouse event
+    // gridWindow: the same or the corresponding non-frozen window
+    int w, h;
+    eventGridWindow->GetSize(&w, &h);
```

Comment:
It looks like we only use those to check if a point is inside the window. If it's really the case, couldn't we just convert the point to screen coordinates using `ScreenToClient()` and then use `GetScreenRect().Contains(point)` to test for this? It looks like it could be simpler and it would definitely be more readable/more "obvious".

Reply:
I will check, but w and/or h can be modified.
The size of the non-frozen window is added if the mouse has started in the frozen part and moved to the non-frozen. That's why I used two variables instead of wxPoint/wxSize.